scanners/trivy/templates/trivy-scan-type.yaml

```diff
@@ -297,4 +297,54 @@ spec:
           serviceAccountName: trivy-k8s
           volumes:
             {{- toYaml .Values.scanner.extraVolumes | nindent 12 }}
-
+---
+apiVersion: "execution.securecodebox.io/v1"
+kind: ScanType
+metadata:
+  name: "trivy-image-sbom{{ .Values.scanner.nameAppend | default ""}}"
+spec:
+  extractResults:
+    type: sbom-cyclonedx
```

Comment:
One possible name inconsistency is here, the scan is named `trivy-image-sbom`, because generating an SBOM in SPDX format is a simple parameter change to the trivy command (`--format spdx-json`), but the output format is named `sbom-cyclonedx` because the parser can (currently) not handle SPDX and neither can the Dependency-Track hook.

So this is a bit of a design question, how far separated do we want to handle SPDX SBOMs in case we add them as well? With completely separated formats/parsers, or integrated under the SBOM name?

Reply:
I think the names for the helm chart "trivy-sbom" and for the result type "sbom-cyclondx" are good.
The scanType could maybe be changed to `trivy-image-sbom-cyclondx` if there comes a need to also support trivy spdx we can add it a second scanType there. (Though then we might get troubles again with having two parsers in one scanner)

I'm fine either way, we can leave it like it is and rename the scanType once / if we actually have a need to also support spdx.

Reply:
Well at this point the names are a bit different, because `trivy-sbom` is its own scanner, and I didn't want to mix up the names too much, so the scanType is called `trivy-sbom-image`, which I think sounds worse than `trivy-image-sbom`, but follows our usual convention of naming scanTypes after `<scanner>-<target>`.

But if you're otherwise fine with that I'd just keep the names like they are for now. Having one SBOM "parser", that can parse both CycloneDX and SPDX SBOMs also doesn't sound too bad to me, you could detect which one it is and then switch in the beginning. Much more than generating an informational finding "SBOM created" doesn't happen anyway. So instead of completely seperating the code probably only the information-extraction functions to fill that one informational finding are going to differ.

Reply:
@J12934  I guess renaming the scanType would be a breaking change? So maybe we should think about which name we want again, otherwise adding SPDX in the future might be a bit akward with scanType names `trivy-sbom-image` and `trivy-sbom-image-spdx` or something like that.

Reply:
Names work for now, can be merged